lib/rspec/matchers/built_in/include.rb

```diff
@@ -13,34 +13,50 @@ def initialize(*expected)
         # @return [Boolean]
         def matches?(actual)
           @actual = actual
-          perform_match(:all?, :all?)
+          if actual.respond_to?(:include?)
+            @divergent_items = excluded_from_actual
```

Comment:
The fact that you only assign this _sometimes_ means this matcher is potentially prone to a bug:

``` ruby
include_one = include(1)
expect([1, 2, 3]).to include_one
expect(Object.new).to include_one
```

On the 2nd use of the matcher, `@divergent_items` will still have the value from the first use of the matcher.  It looks like there isn't actually a bug from this yet since your use of `@divergent_items` in the failure message methods is guarded by a `respond_to` check, but it's a bug waiting to happen if/when that ivar got used for something else.  I'd prefer to see it always assigned.

In addition, the `if actual.respond_to?(:include)` here and for `does_not_match?` is duplicated.  I think we can kill two birds with one stone:

``` ruby
def matches?(actual)
  @actual = actual
  @divergent_items = excluded_from_actual
  @divergent_items.empty?
end

def does_not_match?(actual)
  @actual = actual
  @divergent_items = included_from_actual
  @divergent_items.empty?
end

def excluded_from_actual(...)
  return [] unless actual.respond_to?(:include?)
  # ...
end
```

Thoughts?